avr-hal-generic/src/i2c.rs

```diff
@@ -329,51 +336,110 @@ macro_rules! impl_twi_i2c {
-                //    If the I-bit in SREG and TWIE in TWCR are set, the MCU will jump to the TWI Interrupt Vector. While the TWINT
-                //    Flag is set, the SCL low period is stretched. The TWINT Flag must be cleared by software by writing a logic one
-                //    to it. Note that this flag is not automatically cleared by hardware when executing the interrupt routine. Also note
-                //    that clearing this flag starts the operation of the TWI, so all accesses to the TWI Address Register (TWAR), TWI
-                //    Status Register (TWSR), and TWI Data Register (TWDR) must be complete before clearing this flag.
+                //    This bit is set by hardware when the TWI has finished its current job and
+                //    expects application software response.  If the I-bit in SREG and TWIE in TWCR
+                //    are set, the MCU will jump to the TWI Interrupt Vector. While the TWINT Flag
+                //    is set, the SCL low period is stretched. The TWINT Flag must be cleared by
+                //    software by writing a logic one to it. Note that this flag is not
+                //    automatically cleared by hardware when executing the interrupt routine. Also
+                //    note that clearing this flag starts the operation of the TWI, so all accesses
+                //    to the TWI Address Register (TWAR), TWI Status Register (TWSR), and TWI Data
+                //    Register (TWDR) must be complete before clearing this flag.
                 //
                 //
                 //    Bit 5 – TWSTA: TWI START Condition Bit
                 //
-                //    The application writes the TWSTA bit to one when it desires to become a Master on the 2-wire Serial Bus. The
-                //    TWI hardware checks if the bus is available, and generates a START condition on the bus if it is free. However,
-                //    if the bus is not free, the TWI waits until a STOP condition is detected, and then generates a new START
-                //    condition to claim the bus Master status. TWSTA must be cleared by software when the START condition has
-                //    been transmitted.
+                //    The application writes the TWSTA bit to one when it desires to become a
+                //    Master on the 2-wire Serial Bus. The TWI hardware checks if the bus is
+                //    available, and generates a START condition on the bus if it is free. However,
+                //    if the bus is not free, the TWI waits until a STOP condition is detected, and
+                //    then generates a new START condition to claim the bus Master status. TWSTA
+                //    must be cleared by software when the START condition has been transmitted.
                 //
                 //    NOTE: `twstart` macro token is defined as `twsta` in chips/[CHIP]/src/lib.rs
```

Comment:
The naming difference tripped me up here until I realized twstart === twsta.

Reply:
Ah, this is all a bit messy ... I think we can get away with removing the register name macro tokens entirely like we did for the SPI driver.  Do you want me to do it and then you rebase your code ontop or do you want to take a stab at it yourself?

Reply:
I'll take a run at it. I think I understand the pattern in SPI driver.

Reply:
Essentially, remove this block:

```diff
 #[macro_export]
 macro_rules! impl_twi_i2c {
     (
         $(#[$i2c_attr:meta])*
         pub struct $I2c:ident {
             peripheral: $I2C:ty,
             pins: {
                 sda: $sdamod:ident::$SDA:ident,
                 scl: $sclmod:ident::$SCL:ident,
             },
-            registers: {
-                control: $twcr:ident {
-                    enable: $twen:ident,
-                    ack: $twea:ident,
-                    int: $twint:ident,
-                    start: $twstart:ident,
-                    stop: $twstop:ident,
-                },
-                status: $twsr:ident {
-                    prescaler: $twps:ident,
-                    status: $tws:ident,
-                },
-                bitrate: $twbr:ident,
-                data: $twdr:ident,
-            },
         }
     ) => {
```

and then replace all of the `$...` register/field identifiers with the real ones.  E.g.

```diff
 fn start(
     &mut self,
     addr: u8,
     dir: $crate::i2c::Direction,
 ) -> Result<(), $crate::i2c::Error> {
     // Write start condition
-    self.p.$twcr.write(|w| w
+    self.p.twcr.write(|w| w
-        .$twen().set_bit()
+        .twen().set_bit()
-        .$twint().set_bit()
+        .twint().set_bit()
-        .$twstart().set_bit()
+        .twsta().set_bit()
     );
     self.wait();
```

Reply:
done.